chia/plotting/manager.py

```diff
@@ -0,0 +1,333 @@
+import logging
+import threading
+import time
+import traceback
+from functools import reduce
+from pathlib import Path
+from typing import Any, Callable, Dict, List, Optional, Set, Tuple
+from concurrent.futures.thread import ThreadPoolExecutor
+
+from blspy import G1Element
+from chiapos import DiskProver
+
+from chia.consensus.pos_quality import UI_ACTUAL_SPACE_CONSTANT_FACTOR, _expected_plot_size
+from chia.plotting.util import (
+    PlotInfo,
+    PlotRefreshResult,
+    PlotsRefreshParameter,
+    get_plot_directories,
+    get_plot_filenames,
+    parse_plot_info,
+    stream_plot_info_pk,
+    stream_plot_info_ph,
+)
+from chia.types.blockchain_format.proof_of_space import ProofOfSpace
+from chia.types.blockchain_format.sized_bytes import bytes32
+from chia.wallet.derive_keys import master_sk_to_local_sk
+
+log = logging.getLogger(__name__)
+
+
+class PlotManager:
+    plots: Dict[Path, PlotInfo]
+    plot_filename_paths: Dict[str, Tuple[str, Set[str]]]
+    plot_filename_paths_lock: threading.Lock
+    failed_to_open_filenames: Dict[Path, int]
+    no_key_filenames: Set[Path]
+    farmer_public_keys: List[G1Element]
+    pool_public_keys: List[G1Element]
+    match_str: Optional[str]
+    show_memo: bool
+    open_no_key_filenames: bool
+    last_refresh_time: float
+    refresh_parameter: PlotsRefreshParameter
+    log: Any
+    _lock: threading.Lock
+    _refresh_thread: Optional[threading.Thread]
+    _refreshing_enabled: bool
+    _refresh_callback: Optional[Callable]
```

Comment:
i think now we can remove the optional from here as well, and the None check in line 127

Reply:
Dropping it leads to `error: Cannot assign to a method` by `mypy` for the constructor and `set_refresh_callback` so we need to keep it unless you have a way to make `mypy` happy without it.

Reply:
in that case i would change the if in line 127 to an assert before pushing

Reply:
Cool, changed it :+1: